operator/pkg/model/translation/gateway-api/translator.go

```diff
@@ -105,9 +106,12 @@ func getService(resource *model.FullyQualifiedResource, allPorts []uint32, label
 
 	return &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        shorten(ciliumGatewayPrefix + resource.Name),
-			Namespace:   resource.Namespace,
-			Labels:      mergeMap(map[string]string{owningGatewayLabel: resource.Name}, labels),
+			Name:      shorten(ciliumGatewayPrefix + resource.Name),
+			Namespace: resource.Namespace,
+			Labels: mergeMap(map[string]string{
+				owningGatewayLabel:          resource.Name,
```

Comment:
I believe there is a bug here, we can't directly reference the name of the resource.

I have provided a reproduction step. https://github.com/cilium/cilium/issues/31285

Reply:
interesting, is it due the long name of resource ?